app/driver.go

```diff
@@ -148,11 +148,16 @@ func chooseDriver(kubeClient *clientset.Clientset) (string, error) {
 }
 
 func deployCSIDriver(kubeClient *clientset.Clientset, c *cli.Context, managerImage string) error {
+	namespace := os.Getenv(types.EnvPodNamespace)
+
+	if checked, err := csi.CheckMountPropagationWithPodSpec(kubeClient, managerImage, namespace); err != nil || !checked {
```

Comment:
I thought it should be a switch in the chooseDriver. If mount propagation check failed, we should fall back to Flexvolume (and explicitly tell user in the log what's happened)

Reply:
Also we need to consider if user explicitly choose CSI but we cannot deploy. In this case we should error out instead.